### ISSUE TITLE
tf.keras model fit couldn't support DataFrame interface 

### ISSUE THREAD

**Issue by jack1981** [intel-analytics/analytics-zoo#1898]
Most of time we already used DataFrame to do feature engineering but we have to call  TFDataset.from_rdd interface to transfer the DF to TFDataSet interface and there is no "from_dataframe" function from TFDataSet. And the developer has to go to low level API to transfer rdd to data structures , such as call lambda , np.array etc..
   Will be better experience to support high level Spark data frame interface and simply the function parameters.
       
    trainingDF = genData(tDF, "201312", "201401")
    validationDF = genData(tDF, "201402", "201403")
    model = getKerasModel()
    
    train_rdd = trainingDF.rdd.map(lambda row: (np.array(row["features"]), np.array(int(row["labels"]))))
    validation_rdd = validationDF.rdd.map(lambda row: (np.array(row["features"]), np.array(int(row["labels"]))))
    dataset = TFDataset.from_rdd(train_rdd, features=(tf.float32, [3]), labels=(tf.int32, []), batch_size=20,val_rdd=validation_rdd)
    
    model.fit(dataset,max_epoch,distributed=True)

**Comment by jason-dai** [comment]
@yangw1234 any update?

**Comment by yangw1234** [comment]
@jason-dai currently developing, will submit a pr later today or tomorrow
